verticapy/_utils/_map.py

```diff
@@ -74,3 +74,49 @@ def verticapy_agg_name(key: str, method: Optional[Literal["vertica"]] = "") -> s
         elif key == "std":
             key = "stddev"
     return key
+
+def add_docstring(*args):  # decorator factory function
+    """
+    Constructs and inserts a parameter docstring into the decorated function's existing docstring.
+    The decorator accepts a dictionary of parameter descriptions and then the keys to the parameters
+    of the decorated function. For example:
+
+@add_docstring(PARAMETER_DOCSTRINGS, 'y_true', 'y_score', 'input_relation', 'pos_label')
+
+    The decorator inserts the supplied parameter descriptions inbetween the function's description
+    and the Returns section. For instance, in the following docstring, the above decorator would 
+    insert the parameter descriptions inbetween 'Computes the Confusion Matrix' and 'Returns':
+
+    \'''
+    Computes the Confusion Matrix.
+    
+    Returns
+    -------
+    Array
+        confusion matrix.
+    \'''
+
+    When several functions share the same parameters, this decorator can be used to improve code 
+    readability and doc consistency.
+   """
+    param_defs = args[0]
+    param_docstring = '''
+    Parameters
+    ---------- \n'''
+    for param in args[1:]:
+        param_docstring += param_defs[param] + '\n'
+    param_docstring += '''
+    Returns
+    -------'''
+
+    # actual decorator; its only argument is the decorated function
+    def docstring_decorator(func): 
+        existing_docstring = func.__doc__ if func.__doc__ else ''
+        existing_docstring = existing_docstring.split('''Returns
+    -------''')
+        func.__doc__ = existing_docstring[0] + param_docstring + existing_docstring[1]
+
+        return func
+
+    # @add_docstring(*args) evaluates to the appropriate @docstring_decorator
+    return docstring_decorator
```

Comment:
@oualib 
This is one option for using decorators to reuse parameter descriptions. In this case, you'd create a dictionary of parameter descriptions for each file (see classification.py) and then supply that dictionary and the parameters of the function to the decorator. The decorator would then insert the parameter descriptions between the function description and the listed return value.  
So, you would only need to define each parameter once in the parameter dictionary. Then all you need to do is supply the necessary parameter keys to the decorator.
To use the decorator, you'd just need to import the function add_docstring (didn't where this should go, so put it in _utils._map.py for now)
I don't know how this impacts function runtime, or how it interacts with nested decorators. 
This is my best shot for now. If this is a bad approach, just let me know and I'll review the doc as normal and come back to this sometime in the future.

Reply:
Nice, can you please check the render when you use the 'help' function on one of the function? Is it well aligned?

Reply:
Looks like it renders alright:


![help-printout](https://user-images.githubusercontent.com/40260999/233142063-495188cc-bd86-4899-a55c-9532b3cecd5f.png)

Reply:
Nice!